overlord/overlord.go

```diff
@@ -281,6 +286,37 @@ func (o *Overlord) Loop() {
 	})
 }
 
+// canGoSocketActivated returns true if the main ensure loop can go into
+// "socket-activation" mode. This is only possible once seeding is done
+// and there are no snaps on the system. This is to reduce the memory
+// footprint on e.g. containers.
+func (o *Overlord) canGoSocketActivated() bool {
+	st := o.State()
+	st.Lock()
+	defer st.Unlock()
+
+	// check if there are snaps
+	if n, err := snapstate.NumSnaps(st); err != nil || n > 0 {
+		return false
+	}
+	// check if seeding is done
+	var seeded bool
+	if err := st.Get("seeded", &seeded); err != nil {
+		return false
+	}
+	if !seeded {
+		return false
+	}
+	// check if there are any changes in flight
+	for _, chg := range st.Changes() {
```

Comment:
should we wait also for cleanup, which is a different concept? see checks in Settle about that